File: weather_script.js
const apiKey = "f14a0ca5c7a31440e709d1677124c38e"; // Reemplaza con tu clave de API de OpenWeatherMap
const city = "Madrid"; // Cambia por la ciudad que desees consultar

async function getWeather() {
    const url = `https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${apiKey}&units=metric`;

    try {
        const response = await fetch(url);
        if (!response.ok) {
            throw new Error(`Error: ${response.status} - ${response.statusText}`);
        }

        const data = await response.json();
        console.log(`El clima en ${data.name}:`);
        console.log(`Temperatura: ${data.main.temp}°C`);
        console.log(`Humedad: ${data.main.humidity}%`);
        console.log(`Descripción: ${data.weather[0].description}`);
    } catch (error) {
        console.error("Error al obtener los datos meteorológicos:", error.message);
    }
}

// Llama a la función para obtener los datos meteorológicos
getWeather();

File: weather_script.py
import requests

def get_weather(city, api_key):
    base_url = "http://api.openweathermap.org/data/2.5/weather"
    params = {
        'q': city,
        'appid': api_key,
        'units': 'metric'
    }
    response = requests.get(base_url, params=params)
    return response.json()

def main():
    api_key = "your_api_key_here"
    city = input("Enter city name: ")
    weather_data = get_weather(city, api_key)
    
    if weather_data.get("cod") != 200:
        print("Error:", weather_data.get("message"))
    else:
        print(f"City: {weather_data['name']}")
        print(f"Temperature: {weather_data['main']['temp']}°C")
        print(f"Weather: {weather_data['weather'][0]['description']}")

if __name__ == "__main__":
    main()